diff -ru gcc-4.7.3/config.sub gcc_adapted/config.sub
--- gcc-4.7.3/config.sub	2015-01-03 16:55:55.389796157 +0100
+++ gcc_adapted/config.sub	2015-01-04 16:44:04.921099247 +0100
@@ -1383,6 +1383,9 @@
 	-linux*)
 		os=`echo $os | sed -e 's|linux|linux-gnu|'`
 		;;
+	-universe*)
+		os=-universe
+		;;
 	-sunos5*)
 		os=`echo $os | sed -e 's|sunos5|solaris2|'`
 		;;
Only in gcc_adapted/gcc/config: universe.h
diff -ru gcc-4.7.3/gcc/config.gcc gcc_adapted/gcc/config.gcc
--- gcc-4.7.3/gcc/config.gcc	2015-01-03 16:56:03.293129282 +0100
+++ gcc_adapted/gcc/config.gcc	2015-01-04 16:53:24.324289790 +0100
@@ -526,6 +526,12 @@
 
 # Common parts for widely ported systems.
 case ${target} in
+*-*-universe*)
+   extra_parts="crtbegin.o crtend.o"
+   gas=yes
+   gnu_ld=yes
+   default_use_cxa_atexit=yes
+   ;;
 *-*-darwin*)
   tmake_file="t-darwin ${cpu_type}/t-darwin"
   tm_file="${tm_file} darwin.h"
@@ -934,6 +940,10 @@
 	tm_file="dbxelf.h elfos.h newlib-stdint.h arm/unknown-elf.h arm/elf.h arm/aout.h arm/arm.h"
 	tmake_file="arm/t-arm arm/t-arm-elf"
 	;;
+arm*-*-universe-*)
+	tm_file="dbxelf.h elfos.h newlib-stdint.h arm/unknown-elf.h arm/elf.h arm/aout.h arm/arm.h universe.h"
+	tmake_file="arm/t-arm arm/t-arm-elf"
+	;;
 arm*-wince-pe*)
 	tm_file="arm/semi.h arm/aout.h arm/arm.h arm/coff.h dbxcoff.h arm/pe.h arm/wince-pe.h"
 	tmake_file="arm/t-arm arm/t-wince-pe"
@@ -1245,6 +1255,11 @@
 	gas=yes
 	gnu_ld=yes
 	;;
+i[3-7]86-*-universe*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h universe.h"
+	tmake_file="i386/t-i386elf t-svr4"
+	use_fixproto=yes
+	;;
 i[34567]86-*-linux* | i[34567]86-*-kfreebsd*-gnu | i[34567]86-*-knetbsd*-gnu | i[34567]86-*-gnu* | i[34567]86-*-kopensolaris*-gnu)
 			# Intel 80386's running GNU/*
 			# with ELF format using glibc 2
diff -ru gcc-4.7.3/libgcc/config.host gcc_adapted/libgcc/config.host
--- gcc-4.7.3/libgcc/config.host	2015-01-03 16:55:55.553129486 +0100
+++ gcc_adapted/libgcc/config.host	2015-01-04 16:57:23.937616788 +0100
@@ -379,6 +379,10 @@
 	tmake_file="$tmake_file arm/t-arm arm/t-elf t-softfp-sfdf t-softfp-excl arm/t-softfp t-softfp"
 	extra_parts="$extra_parts crti.o crtn.o"
 	;;
+arm*-*-universe-*)
+	tmake_file="$tmake_file arm/t-arm arm/t-elf t-softfp-sfdf t-softfp-excl arm/t-softfp t-softfp"
+	extra_parts="crtbegin.o crtend.o"
+	;;
 arm*-wince-pe*)
 	tmake_file="$tmake_file arm/t-arm arm/t-wince-pe"
 	;;
@@ -1138,6 +1142,10 @@
 m32c-*-elf*|m32c-*-rtems*)
 	tmake_file="$tmake_file m32c/t-m32c"
  	;;
+i[3-7]86-*-universe*)
+	extra_parts="crtbegin.o crtend.o"
+	tmake_file="$tmake_file i386/t-crtstuff"
+	;;
 mep*-*-*)
 	tmake_file="mep/t-mep t-fdpbit"
 	extra_parts="crtbegin.o crtend.o"
